### PR TITLE
avocado tags extended to kernel and memory tests

### PR DESCRIPTION
**Pull request by abdhaleegit**
tags are added to kernel and memory tests to filter/list/run tests based on the tags

listing/running only tests that can run as regular users
# avocado list --filter-by-tags=-privileged kernel
INSTRUMENTED kernel/posixtest.py:Posixtest.test
INSTRUMENTED kernel/rmaptest.py:Rmaptest.test

# avocado list --filter-by-tags=-privileged memory
INSTRUMENTED memory/eatmemory.py:eatmemory.test
INSTRUMENTED memory/stress-ng.py:stressng.test

Listing/running tests that require a Power architecture:
# avocado list --filter-by-tags=power kernel
INSTRUMENTED kernel/kselftest.py:kselftest.test

# avocado list --filter-by-tags=power  memory
INSTRUMENTED memory/libhugetlbfs.py:libhugetlbfs.test
INSTRUMENTED memory/transparent_hugepages.py:Thp.test
INSTRUMENTED memory/transparent_hugepages_defrag.py:Thp_Defrag.test
INSTRUMENTED memory/transparent_hugepages_swapping.py:Thp_Swapping.test

list component specific filer
# avocado list --filter-by-tags=kernel kernel
INSTRUMENTED kernel/kselftest.py:kselftest.test
INSTRUMENTED kernel/posixtest.py:Posixtest.test
INSTRUMENTED kernel/tlbflush.py:Tlbflush.test

# avocado list --filter-by-tags=power,privileged  memory
INSTRUMENTED memory/libhugetlbfs.py:libhugetlbfs.test
INSTRUMENTED memory/transparent_hugepages.py:Thp.test
INSTRUMENTED memory/transparent_hugepages_defrag.py:Thp_Defrag.test
INSTRUMENTED memory/transparent_hugepages_swapping.py:Thp_Swapping.test

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@abdhaleegit please rebase with having valid commit message it make travis happy :)

File: memory/libhugetlbfs.py [hunk 1]
```diff
@@ -33,6 +33,12 @@
 
 class libhugetlbfs(Test):
 
+    """
+    libhugetlbfs: libhugetlbfs is a library which provides easy
+    access to huge pages of memory. test to excersize libhugetlbfs library
+
+    :avocado: tags=memory,power,privileged
```
Comment (PraveenPenguin) [inline memory/libhugetlbfs.py]: i think this test not only power specific

File: memory/transparent_hugepages.py [hunk 2]
```diff
@@ -33,6 +33,8 @@ class Thp(Test):
     '''
     The test enables THP and stress the system using dd load
     and verifies whether THP has been allocated for usage or not
+
+    :avocado: tags=memory,power,privileged
```
Comment (PraveenPenguin) [inline memory/transparent_hugepages.py]: this also

File: memory/transparent_hugepages_defrag.py [hunk 3]
```diff
@@ -37,6 +37,8 @@ class Thp_Defrag(Test):
     '''
     Defrag test enables THP and fragments the system memory using dd load
     and turns on THP defrag and checks whether defrag occured.
+
+    :avocado: tags=memory,power,privileged
```
Comment (PraveenPenguin) [inline memory/transparent_hugepages_defrag.py]: here as well

File: memory/transparent_hugepages_swapping.py [hunk 4]
```diff
@@ -33,6 +33,8 @@ class Thp_Swapping(Test):
 
     '''
     The test fills out the total avl memory and tries to swap the thp out.
+
+    :avocado: tags=memory,power,privileged
```
Comment (PraveenPenguin) [inline memory/transparent_hugepages_swapping.py]: here as well

File: kernel/kselftest.py [hunk 5]
```diff
@@ -32,6 +32,8 @@ class kselftest(Test):
 
     :see: https://www.kernel.org/doc/Documentation/kselftest.txt
     :source: https://github.com/torvalds/linux/archive/master.zip
+
+    :avocado: tags=kernel,power,privileged
```
Comment (PraveenPenguin) [inline kernel/kselftest.py]: this test also not specific to power only

File: memory/libhugetlbfs.py (hunk 1 above)
Comment (abdhaleegit) [inline memory/libhugetlbfs.py]: yes, you are right

File: memory/transparent_hugepages.py (hunk 2 above)
Comment (abdhaleegit) [inline memory/transparent_hugepages.py]: but the test is designed for powerpc and not verified on x86

Comment (PraveenPenguin) [inline memory/transparent_hugepages.py]: no i don't think so @santhosh93 can you please comment ?

File: kernel/kselftest.py (hunk 5 above)
Comment (abdhaleegit) [inline kernel/kselftest.py]: all the test inside tools/testing/selftest/powerpc  are specific to power like perf, pmu, ebb, radix mmu,etc

File: memory/transparent_hugepages.py (hunk 2 above)
Comment (abdhaleegit) [inline memory/transparent_hugepages.py]: checked with santhosh, it is generic.. will change that

File: memory/transparent_hugepages_defrag.py (hunk 3 above)
Comment (abdhaleegit) [inline memory/transparent_hugepages_defrag.py]: changed

File: memory/transparent_hugepages_swapping.py (hunk 4 above)
Comment (abdhaleegit) [inline memory/transparent_hugepages_swapping.py]: changed

**Comment by PraveenPenguin** [comment]
@abdhaleegit sure can you please push the changes

File: kernel/kselftest.py (hunk 5 above)
Comment (PraveenPenguin) [inline kernel/kselftest.py]: no still i think it is not power specific

**Pull request closed without merging**